engine/src/main.cpp

```diff
@@ -117,6 +117,11 @@ boost::shared_ptr<VSFileSystem::console_log_sink> VSFileSystem::pConsoleLogSink
 
 //false if command line option --net is given to start without network
 static bool ignore_network = true;
+// legacy_mode determines whether the application should require the data directory to be specified on the command-line or not
+// true - no parameter required, application should start regardless of where it's called from
+// false - parameter required, application should only start if specified
+// NOTE: This is tied to the executable name. `vegastrike` -> true; `vegastrike-engine` -> false
+bool legacy_mode;
```

Comment:
Hmm. This variable name is OK, but I think we can do better. How about something like `legacy_data_dir_mode`?

Reply:
@stephengtuggy I was intending for it to be for anything that depended on the legacy behavior; the data directory is certainly one of them. Is it the only one?

Reply:
@BenjamenMeyer I can't think of any others at the moment.

Reply:
@Loki1950 are there others we may run into?

Reply:
I think we have covered them all

Reply:
okay - I'll put up a small change to rename it.

Reply:
updated